From db0381456367db33b0e433051c141af4c93ae1d0 Mon Sep 17 00:00:00 2001
From: siddhesh <siddhesh>
Date: Wed, 26 Sep 2012 02:06:51 +0000
Subject: [PATCH] gdb/ChangeLog:

	* breakpoint.c (invalidate_bp_value_on_memory_change): Expand
	parameter LEN to ssize_t.

gdb/doc/ChangeLog:

	observer.texi (memory_changed): Expand parameter LEN to ssize_t.
---
 gdb/ChangeLog         |    5 +++++
 gdb/breakpoint.c      |    2 +-
 gdb/doc/ChangeLog     |    4 ++++
 gdb/doc/observer.texi |    2 +-
 4 files changed, 11 insertions(+), 2 deletions(-)

Index: gdb-7.2/gdb/breakpoint.c
===================================================================
--- gdb-7.2.orig/gdb/breakpoint.c
+++ gdb-7.2/gdb/breakpoint.c
@@ -11005,7 +11005,7 @@ show_breakpoint_cmd (char *args, int fro
    GDB itself.  */
 
 static void
-invalidate_bp_value_on_memory_change (CORE_ADDR addr, int len,
+invalidate_bp_value_on_memory_change (CORE_ADDR addr, ssize_t len,
 				      const bfd_byte *data)
 {
   struct breakpoint *bp;
Index: gdb-7.2/gdb/doc/observer.texi
===================================================================
--- gdb-7.2.orig/gdb/doc/observer.texi
+++ gdb-7.2/gdb/doc/observer.texi
@@ -218,7 +218,7 @@ The inferior @var{inf} has been removed 
 This method is called immediately before freeing @var{inf}.
 @end deftypefun
 
-@deftypefun void memory_changed (CORE_ADDR @var{addr}, int @var{len}, const bfd_byte *@var{data})
+@deftypefun void memory_changed (CORE_ADDR @var{addr}, ssize_t @var{len}, const bfd_byte *@var{data})
 Bytes from @var{data} to @var{data} + @var{len} have been written
 to the current inferior at @var{addr}.
 @end deftypefun
